docs/OptionSettingsItems-input-validation.md

```diff
@@ -0,0 +1,193 @@
+# OptionSettingItems Input Validation Design Document
+_DOTNET-4952_
+
+## Summary
+Need mechanism to indicate an `OptionSettingItem` is required (ie non empty value) or has additional input validations:
+ - less than 64 characters
+ - no whitespace characters
+ - no non-ascii characters
+ - value is from list of acceptable values
+ - is positive number
+ - is number within bounds
+
+Deployment should be blocked if there are any OptionSettingItems with invalid values.
+
+## Proposed Design
+
+Unqiue validation classes will be created and will implement a new interface:
+
+```csharp
+public interface IOptionSettingItemValidator
+{
+    OptionSettingItemValidationResult Validate(object input);
+}
+
```

Comment:
Reason behind not using just a lambda which does validation then using full classes that implements IOptionSettingItemValidator?

Reply:
these classes contain some configuration (ie `ValidationFailedMessage`) and need to be serializable.